src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -562,6 +562,7 @@ protected function isDecimalCast($cast)
      */
     public function setAttribute($key, $value)
     {
+        $key = (string) $key;
```

Comment:
The phpdoc says that the key is already a string, so any code not passing a string to this function needs to be fixed, instead, and then this line removed.

Reply:
Thanks. Agree with that. I guess I should just cast `string` at the `__set` method in `Model` class like the following, right? 
```
public function __set($key, $value)
{
      $this->setAttribute((string) $key, $value);
}
```

Reply:
Although, there are a lot of other `setAttribute` method calls. Do you think all of them should cast `string` on their keys too?

Reply:
@GrahamCampbell  I've updated the PR by changing the implementation as Suggested in the Issue. As casting the $key to `string` in every `setAttribute` call doesn't seem like a good solution to me.

Reply:
> Thanks. Agree with that. I guess I should just cast string at the __set method in Model class like the following, right?

Any dynamic call to that method will already provide a non-empty string as the key. No need to cast.